### DISCUSSION TITLE
On passing a Connection object with an open transaction

### DISCUSSION THREAD

**Discussion by bourreta** [jobrunr/jobrunr#492] [Q&A]
In order to ensure a job is saved (enqueued) within a transaction, is there a way to pass a Connection object with an open transaction? Or, is the only way to do this by using `@Transactional`, which retrieves the connection object from a ThreadLocal?

**Comment by rdehuyss** [comment]
Well, support for `@Transactional` comes out-of-the-box with JobRunr Pro :-). Or you want to integrate with a different framework or do you have very specific transaction needs? I'm just trying to understand what you want to achieve.

**Reply by bourreta** [reply]
Yes… I am not using `@Transactional` and just use the Connection object with auto commit set to false and want to control when the job is committed to the DB along with my other writes.

**Reply by rdehuyss** [reply]
I'm afraid this is not possible in the free version of JobRunr but that is possible in the Pro version - I use the same principle for adding the `@Transactional` support and you could use that too. You're also the first one asking for this kind of feature and I spent quite a lot of time on implementing this nicely, that's why it's not available in the free version.

Drop me a note via hello@jobrunr.io if you're interested in a trial of the Pro version.
